### PR TITLE
Fix llvm37 building on macOS Sierra/El Capitan

### PR DESCRIPTION
**Pull request by EtherTyper**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----

@MikeMcQuaid https://github.com/ethereum/webthree-umbrella/issues/614#issuecomment-231549573

### CODE REVIEW

File: llvm37.rb [hunk 1]
```diff
@@ -155,6 +155,10 @@ def install
     clang_buildpath = buildpath/"tools/clang"
     libcxx_buildpath = buildpath/"projects/libcxx"
     libcxxabi_buildpath = buildpath/"libcxxabi" # build failure if put in projects due to no Makefile
+    
+    inreplace libcxxbuilpath/"include/string",
+      "basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)" # Before
+      "basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a) basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a) noexcept(is_nothrow_copy_constructible<allocator_type>::value)"
 
     clang_buildpath.install resource("clang")
     libcxx_buildpath.install resource("libcxx")
```
Comment (MikeMcQuaid) [inline llvm37.rb]: Put the inreplace after this line and the files will have been extracted.

**Comment by EtherTyper** [comment]
@MikeMcQuaid Hmmm... Any idea why it's failing?

**Comment by MikeMcQuaid** [comment]
Still failing to build unfortunately: http://bot.brew.sh/job/Homebrew%20Versions%20Pull%20Requests/1480/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_homebrew_versions_llvm37/

The `audit` failure is from trailing whitespace: http://bot.brew.sh/job/Homebrew%20Versions%20Pull%20Requests/1480/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/audit_homebrew_versions_llvm37/

Have you managed to build this successfully on your local machine?

**Comment by EtherTyper** [comment]
@MikeMcQuaid Nope. I have no idea why the patch just will not apply itself. Can you check how I did it?

**Comment by MikeMcQuaid** [comment]
@EtherTyper Looks like it may have applied but just isn't sufficient?

**Comment by MikeMcQuaid** [comment]
If you can't test this locally then trying to do it on CI alone may be painful.

**Comment by EtherTyper** [comment]
Yeah... But, the install works when I perform the same edit on the downloaded tarball for libc++ and just manually edit the SHA256 hash to match in the formula, on llvm37.

**Comment by EtherTyper** [comment]
On second thought, you're right. I screwed up the patch.

**Comment by EtherTyper** [comment]
@MikeMcQuaid Why was the build aborted this time?

**Comment by ilovezfs** [comment]
@EtherTyper that was me. It's triggered now.

**Comment by EtherTyper** [comment]
@MikeMcQuaid IT'S BUILDING! Maybe you can squash-merge into a commit that fits naming guidelines, like `llvm37: fix Sierra/El Capitan building` or something.

**Comment by MikeMcQuaid** [comment]
Done, great work here! Want to try and do the same for any other LLVM versions?

**Comment by EtherTyper** [comment]
Okay!

**Comment by EtherTyper** [comment]
@MikeMcQuaid I'll whip up a pull request that fixes all of the following:
```bash
Elis-MBP:llvmtesting ethertyper$ for i in $( ls ); do echo $i $( cat $i/include/string | grep "basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)"); done
libcxx-3.3.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
libcxx-3.4.2.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
libcxx-3.5.1.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
libcxx-3.6.2.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
libcxx-3.7.1.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
libcxx-3.8.0.src basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
```

**Pull request closed without merging**